ee/tabby-webserver/src/service/background_job/third_party_integration.rs

```diff
@@ -140,18 +140,23 @@ impl SchedulerGithubGitlabJob {
         stream! {
             let mut count = 0;
             let mut num_updated = 0;
-            for await (updated_at, doc) in issue_stream.chain(pull_stream) {
-                if index.add(updated_at, doc).await {
+            let mut num_deleted = 0;
+            for await (state, doc) in issue_stream.chain(pull_stream) {
+                let id = &doc.id().to_owned();
+                if !state.should_clean && index.add(state.updated_at, doc).await {
```

Comment:
```suggestion
                if !index.add(state, doc).await {
```

let indexer to decide whether to delete document based on `state`